test/validation/api/ipsec/ipsec.c

```diff
@@ -998,6 +1070,26 @@ int ipsec_check_out(const ipsec_test_part *part, odp_ipsec_sa_t sa,
 		} else {
 			/* IPsec packet */
 			CU_ASSERT_EQUAL(0, odp_ipsec_result(&result, pkto[i]));
+			if (part->flags.test_sa_expiry && !sa_expiry_notified) {
+				if (part->out[i].sa_expiry == IPSEC_TEST_EXPIRY_SOFT_PKT) {
+					if (result.status.warn.soft_exp_packets)
+						sa_expiry_notified = true;
+				} else if (part->out[i].sa_expiry == IPSEC_TEST_EXPIRY_SOFT_BYTE) {
+					if (result.status.warn.soft_exp_bytes)
+						sa_expiry_notified = true;
+				} else if (part->out[i].sa_expiry == IPSEC_TEST_EXPIRY_HARD_PKT) {
+					if (result.status.error.hard_exp_packets) {
```

Comment:
It would be good to check that after certain point the hard expiry error is in every packet. Now the test would pass even if after hard limit expiration (not after the first error packet, since that may become a bit earlier) some later packets would still be processed without errors, which would be a serious problem.

Reply:
Ack. Updated the test to continue sending test packets even after hitting first error packet  and check for the corresponding error bits in the result.